lib/GenomeFileUtil/core/GenbankToGenome.py

```diff
@@ -84,53 +98,121 @@ def __init__(self, config):
     def messages(self):
         return "\n".join(self._messages)
 
-    def refactored_import(self, ctx, params):
-        # 1) validate parameters and extract defaults
-        self.validate_params(params)
-
-        # 2) construct the input directory staging area
-        input_directory = self.stage_input(params)
-
-        # 3) update default params
-        self.default_params.update(params)
-        params = self.default_params
-        self.generate_parents = params.get('generate_missing_genes')
-        self.generate_ids = params.get('generate_ids_if_needed')
-        if params.get('genetic_code'):
-            self.code_table = params['genetic_code']
-
-        # 4) Do the upload
-        files = self._find_input_files(input_directory)
-        consolidated_file = self._join_files_skip_empty_lines(files)
-        genome = self.parse_genbank(consolidated_file, params)
-        if params.get('genetic_code'):
-            genome["genetic_code"] = params['genetic_code']
-
-        result = self.gi.save_one_genome({
-            'workspace': params['workspace_name'],
-            'name': params['genome_name'],
-            'data': genome,
-            "meta": params['metadata'],
-        })
-        ref = f"{result['info'][6]}/{result['info'][0]}/{result['info'][4]}"
-        logging.info(f"Genome saved to {ref}")
-
-        # 5) clear the temp directory
-        shutil.rmtree(input_directory)
-
-        # 6) return the result
-        info = result['info']
-        details = {
-            'genome_ref': ref,
-            'genome_info': info
-        }
-
+    def refactored_import(self, params):
+        print('validating parameters')
+        mass_params = self._set_up_single_params(params)
+        return self._refactored_import_mass(mass_params)[0]
+
+    def refactored_import_mass(self, params):
```

Comment:
`refactored_import_mass` isn't a good name for a function, as the `refactored` part doesn't give any useful information to the user other than at some point the function was rewritten in some way. The function name should describe what the function currently does, not its history.

Reply:
👍

Reply:
:+1: